### DISCUSSION TITLE
Is there any way to dynamically add/remove columns from DetailsList?

### DISCUSSION THREAD

**Discussion by arpost1** [microsoft/powercat-creator-kit#259] [Q&A]
Greetings, all! Been experimenting with the DetailsList and have run into a roadblock but wanted to be sure there wasn't a solution before I submit a feature request. I have a scenario in which I need to add/remove columns from the Details list dynamically. Anyone know of a way to:

1. **~~Make the Columns(Items) property dynamic?~~** Figured out how to do this with Parse JSON; the field cards remain the issue.
2. Add/remove fields?

Or is this just a pipe dream?

**Comment by yaseralhosani** [comment]
Hi @arpost1,

I'm not sure it will work, but you could able to do it with a tag picker, collection, and some formula in DetailsList.columns_Items inside block {}.

I'm able to shift the column like below:
```
{
      ColName: If(true,"ColumnName1","ColumnName2"),
      ColDisplayName:  If(true,"Name1","Name2")
}
```
